modules/vmfcore/src/metadatastream.cpp

```diff
@@ -836,5 +846,75 @@ void MetadataStream::convertFrameIndexToTimestamp(
         timestamp = Metadata::UNDEFINED_TIMESTAMP, duration = Metadata::UNDEFINED_DURATION;
 }
 
+void MetadataStream::notifyStat(StatAction::Type action, std::shared_ptr< Metadata > spMetadata)
+{
+    std::for_each( m_stats.begin(), m_stats.end(), [&]( Stat* stat )
+    {
+        stat->notify(action, spMetadata);
+    });
+}
+
+void MetadataStream::addStat(const std::string& name, const std::vector< StatField >& fields, StatUpdateMode::Type updateMode)
+{
+    auto it = std::find_if(m_stats.begin(), m_stats.end(), [&]( const Stat* stat)->bool
+    {
+        return stat->getName() == name;
+    });
+
+    if (it != m_stats.end())
+    {
+        VMF_EXCEPTION(IncorrectParamException, "Statistics object already exists: '" + name + "'");
+    }
+
+    Stat* stat = new Stat(name, fields, updateMode);
+    m_stats.push_back(stat);
```

Comment:
if using `m_stats.emplace_back(name, fields, updateMode)` - it's possible to use `vector<Stat>`, not pointers.